pkg/cmd/admin/router/router.go

```diff
@@ -642,12 +654,12 @@ func RunCmdRouter(f *clientcmd.Factory, cmd *cobra.Command, out io.Writer, cfg *
 		if cfg.SecretsAsEnv {
 			env.Add(app.Environment{"DEFAULT_CERTIFICATE": string(defaultCert)})
 		} else {
-			// TODO: make --credentials create secrets and bypass service account
 			env.Add(app.Environment{"DEFAULT_CERTIFICATE_PATH": defaultCertificatePath})
 		}
 	}
-
-	secrets, volumes, mounts, err := generateSecretsConfig(cfg, kClient, namespace, defaultCert)
+	env.Add(app.Environment{"DEFAULT_CERTIFICATE_DIR": defaultCertificateDir})
```

Comment:
I guess my question comes down to why wouldn't DEFAULT_CERTIFICATE_PATH be sufficient - why do we need the new env var in order to get this behavior (now that we are auto generating a cert)?